RobotFramework_UDS/DiagnosticServices.py

```diff
@@ -261,14 +262,42 @@ def get_did_codec(self, service_id):
   A dictionary where the keys are DIDs
         """
         did_codec = {}
-
-        diag_services = self.diag_layer.service_groups[service_id]
+        diag_services = self.ecus.service_groups[service_id]
         for diag_service in diag_services:
-            did = diag_service.request.parameters[1].coded_value
-            did_codec[did] = PDXCodec(diag_service)
+            did = self.get_id_base_on_parameters_type(diag_service, sub_services)
+            if isinstance(did, int):
+                did_codec[did] = PDXCodec(diag_service)
+            elif isinstance(did, dict):
+                for data_id in list(did.keys()):
+                    did_codec[data_id] = PDXCodec(diag_service)
 
         return did_codec
 
+    @staticmethod
+    def get_id_base_on_parameters_type(main_service, dict_sub_services = None):
```

Comment:
- make this method name as generic such as `get_param_value`, not only to get did.
- it can also be extended to support other parameter types (if required) than `CODED-CONST` and `TABLE-KEY`